aten/src/ATen/native/ReplicationPadding.cpp

```diff
@@ -1,150 +1,143 @@
 #include "ATen/ATen.h"
 #include "ATen/NativeFunctions.h"
+#include <algorithm>
 
 namespace at {
 namespace native {
 
 namespace {
 template <typename scalar_t>
 static void replication_pad1d_out_frame(
-    scalar_t *input_p, scalar_t *output_p,
+    scalar_t *input_data, scalar_t *output_data,
     long nslices,
     long iwidth,
     long owidth,
-    int pad_l, int pad_r)
+    int pad_l, int pad_r,
+    int nbatch)
 {
-  int iStartX = fmax(0, -pad_l);
-  int oStartX = fmax(0, pad_l);
+  int iStartX = std::max(0, -pad_l);
+  int oStartX = std::max(0, pad_l);
 
-  long k, ip_x;
-#pragma omp parallel for private(k, ip_x)
-  for (k = 0; k < nslices; k++)
+  long p, k, ip_x;
+#pragma omp parallel for private(p, k, ip_x)
```

Comment:
Meh, there is a downside to doing it this way, which is that you are now calling into an OpenMP parallelized loop even when batch size is 1, which comes at a cost. I don't remember if OpenMP is smart enough not to ship the work to a worker thread when this happens but some quick eyeballing of the assembly suggests the answer is no: https://godbolt.org/z/hAV-gF

So, if you wanted to defend this choice, you would have to show via benchmarks that this doesn't actually matter. But in your posted CPU timeits things DID get slower. So I don't want to allow this pessimization for no good reason.

Reply:
(The alternate way, is to continue to have two codepaths for the batch and non-batch case, and have one extra little stub function for the batch case which sets up the OMP parallelization before calling the non-batch function.)

See also discussion in #15480.